### ISSUE TITLE
Suggesting Edit: Node-to-Node Messaging

### ISSUE THREAD

**Issue by simwilso** [holochain/docs-pages#174]
HDK reference URL at the bottom is broken.
It should be this instead:
https://docs.rs/hdk/0.0.42-alpha5/hdk/api/fn.send.html

Also it would be helpful to have the HDK document updated with some example code showing how to call the function using the newer macros i.e:
```
#[zome_fn("hc_public")]
    fn send(_to_agent: Address, _message: String) -> ZomeApiResult<String> {
        hdk::send(
            _to_agent,
            _message,
            300000.into(),
        )
    }
```
and also some information in the HDK around the receive macro implementation i.e. receiver zome function:
```
#[receive]
    pub fn receive(from: Address, msg_json: JsonString) -> String {
        ...
        }
```

Page URL: http://developer.holochain.org/docs/concepts/8_node_to_node_messaging/

**Comment by pdaoust** [comment]
Hi @simwilso you've touched on a regular annoyance of mine too -- we need much better documentation on the `receive` callback (and validator callbacks as well). In general the HDK documentation needs more code samples, but callbacks are particularly in need.

**Comment by pdaoust** [comment]
The broken-URL issue is fixed by #195. Leaving open for the recommendations on improving HDK docks.
